### PR TITLE
[OM-82269]: Replace VMPM_ACCESS commodity with LABEL commodity for no…

### PR DESCRIPTION
**Pull request by amd-ibm**
…deSelector.

### CODE REVIEW

File: pkg/discovery/dtofactory/node_entity_dto_builder.go [hunk 1]
```diff
@@ -228,6 +229,20 @@ func (builder *nodeEntityDTOBuilder) getNodeCommoditiesSold(node *api.Node, clus
...
+		labelComm, err := sdkbuilder.NewCommodityDTOBuilder(proto.CommodityDTO_LABEL).
+			Key(label).
+			Capacity(labelCommodityDefaultCapacity).
+			Create()
+		if err != nil {
+			return nil, isAvailableForPlacement, err
+		}
+		glog.V(5).Infof("Adding label commodity for Node %s with key : %s", node.Name, label)
+		commoditiesSold = append(commoditiesSold, labelComm)
+	}
```
Comment (ading1977) [inline pkg/discovery/dtofactory/node_entity_dto_builder.go:244]: We need to remove the logic to create `VMPM_ACCESS` commodity for labels above

File: pkg/discovery/dtofactory/pod_entity_dto_builder.go [hunk 2]
```diff
@@ -338,6 +338,16 @@ func (builder *podEntityDTOBuilder) getPodCommoditiesSold(
...
+	// label commodity
+	labelComm, err := sdkbuilder.NewCommodityDTOBuilder(proto.CommodityDTO_LABEL).
+		Key(string(pod.UID)).
+		Capacity(labelCommodityDefaultCapacity).
+		Create()
+	if err != nil {
+		return nil, err
+	}
+	commoditiesSold = append(commoditiesSold, labelComm)
+
```
Comment (ading1977) [inline pkg/discovery/dtofactory/pod_entity_dto_builder.go:350]: We should not create `LABEL`  sold commodities for pods. We create `VMPM_ACCESS` commodities between pod sold and container bought to bind the container to the pod, such that it cannot be moved out of the pod in Market.

File: pkg/discovery/dtofactory/pod_entity_dto_builder.go [hunk 3]
```diff
@@ -388,6 +398,19 @@ func (builder *podEntityDTOBuilder) getPodCommoditiesBought(
...
+		selector := key + "=" + value
+		labelComm, err := sdkbuilder.NewCommodityDTOBuilder(proto.CommodityDTO_LABEL).
+			Key(selector).
+			Create()
+		if err != nil {
+			return nil, err
+		}
+		glog.V(5).Infof("Adding label commodity for Pod %s with key : %s", pod.Name, selector)
+		commoditiesBought = append(commoditiesBought, labelComm)
+	}
```
Comment (ading1977) [inline pkg/discovery/dtofactory/pod_entity_dto_builder.go:412]: Similar to node, we need to remove the logic to create `VMPM_ACCESS` bought commodity for labels above.

**Review by ading1977 — CHANGES_REQUESTED** [review]

File: pkg/discovery/dtofactory/container_dto_builder.go [hunk 4]
```diff
@@ -317,6 +317,15 @@ func (builder *containerDTOBuilder) getCommoditiesBought(podId, containerName, c
...
 
+	// 3. Label
+	labelComm, err := sdkbuilder.NewCommodityDTOBuilder(proto.CommodityDTO_LABEL).
+		Key(podId).
+		Create()
+	if err != nil {
+		return nil, err
+	}
+	result = append(result, labelComm)
+
```
Comment (ading1977) [inline pkg/discovery/dtofactory/container_dto_builder.go:328]: Similar to pod sold, we should not create container pod `LABEL` commodity.

File: pkg/discovery/dtofactory/node_entity_dto_builder.go (hunk 1 above)
Comment (amd-ibm) [reply pkg/discovery/dtofactory/node_entity_dto_builder.go]: Removed.

File: pkg/discovery/dtofactory/pod_entity_dto_builder.go (hunk 3 above)
Comment (amd-ibm) [reply pkg/discovery/dtofactory/pod_entity_dto_builder.go]: Removed.

**Pull request merged**